Sources/PackageLoading/ManifestLoader.swift

```diff
@@ -801,69 +817,91 @@ public final class ManifestLoader: ManifestLoaderProtocol {
+                let hFile: Int = _get_osfhandle(_fileno(jsonOutputFileDesc))
+                cmd += ["-handle", "\(String(hFile, radix: 16))"]
+    #else
+                cmd += ["-fileno", "\(fileno(jsonOutputFileDesc))"]
+    #endif
+
+                do {
+                    let packageDirectory = manifestPath.parentDirectory.pathString
+                    let contextModel = ContextModel(packageDirectory: packageDirectory)
+                    cmd += ["-context", try contextModel.encode()]
+                } catch {
+                    return completion(.failure(error))
+                }
 
-            // Return now if there was an error.
-            if runResult.exitStatus != .terminated(code: 0) {
-                result.errorOutput = runOutput
-                return
-            }
+                // If enabled, run command in a sandbox.
+                // This provides some safety against arbitrary code execution when parsing manifest files.
+                // We only allow the permissions which are absolutely necessary.
+                if self.isManifestSandboxEnabled {
+                    let cacheDirectories = [self.databaseCacheDir, moduleCachePath].compactMap{ $0 }
+                    let strictness: Sandbox.Strictness = toolsVersion < .v5_3 ? .manifest_pre_53 : .default
+                    cmd = Sandbox.apply(command: cmd, writableDirectories: cacheDirectories, strictness: strictness)
+                }
 
-            // Read the JSON output that was emitted by libPackageDescription.
-            guard let jsonOutput = try localFileSystem.readFileContents(jsonOutputFile).validDescription else {
-                throw StringError("the manifest's JSON output has invalid encoding")
+                // Run the compiled manifest.
+                var environment = ProcessEnv.vars
+    #if os(Windows)
+                let windowsPathComponent = runtimePath.pathString.replacingOccurrences(of: "/", with: "\\")
+                environment["Path"] = "\(windowsPathComponent);\(environment["Path"] ?? "")"
+    #endif
+    
+                Process.popen(arguments: cmd, environment: environment) { result in
+                    defer { cleanupTmpDir(tmpDir) }
+                    fclose(jsonOutputFileDesc)
+                    
+                    do {
+                        let runResult = try result.get()
+                        if let runOutput = try (runResult.utf8Output() + runResult.utf8stderrOutput()).spm_chuzzle() {
+                            // Append the runtime output to any compiler output we've received.
+                            evaluationResult.compilerOutput = (evaluationResult.compilerOutput ?? "") + runOutput
+                        }
+
+                        // Return now if there was an error.
+                        if runResult.exitStatus != .terminated(code: 0) {
+                            // TODO: should this simply be an error?
```

Comment:
Conceptually it seems better for it to be an error, as long as we pass back any diagnostics and other relevant information.  You might want to pass back the whole evaluation result, which I think has the compiler invocation etc.